runtimes/peregrine/src/lib.rs

```diff
@@ -969,6 +976,61 @@ impl pallet_proxy::Config for Runtime {
 	type WeightInfo = weights::pallet_proxy::WeightInfo<Runtime>;
 }
 
+parameter_types! {
+	pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+}
+
+pub type KiltToEKiltSwitchPallet = pallet_asset_switch::Instance1;
+impl pallet_asset_switch::Config<KiltToEKiltSwitchPallet> for Runtime {
+	type AccountIdConverter = AccountId32ToAccountId32JunctionConverter;
+	type AssetTransactor = FungiblesAdapter<
+		Fungibles,
+		MatchesSwitchPairXcmFeeFungibleAsset<Runtime, KiltToEKiltSwitchPallet>,
+		LocationToAccountIdConverter,
+		AccountId,
+		NoChecking,
+		CheckingAccount,
+	>;
+	type FeeOrigin = EnsureRoot<AccountId>;
+	type LocalCurrency = Balances;
+	type PauseOrigin = EnsureRoot<AccountId>;
+	type RuntimeEvent = RuntimeEvent;
+	type SubmitterOrigin = EnsureSigned<AccountId>;
+	type SwitchHooks = asset_switch::RestrictswitchDestinationToSelf;
+	type SwitchOrigin = EnsureRoot<AccountId>;
+	type WeightInfo = weights::pallet_asset_switch::WeightInfo<Runtime>;
+	type XcmRouter = XcmRouter;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = crate::benches::CreateFungibleForAssetSwitchPool1;
+}
+
+// No deposit is taken since creation is permissioned. Only the root origin can
+// create new assets, and the owner will be the treasury account.
+impl pallet_assets::Config for Runtime {
+	type ApprovalDeposit = ConstU128<0>;
+	type AssetAccountDeposit = ConstU128<0>;
+	type AssetDeposit = ConstU128<0>;
+	type AssetId = Location;
+	type AssetIdParameter = Location;
+	type Balance = u128;
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = runtime_common::asset_switch::NoopBenchmarkHelper;
+	type CallbackHandle = ();
+	type CreateOrigin = AsEnsureOriginWithArg<EnsureRootAsTreasury<Runtime>>;
+	type Currency = Balances;
+	type Extra = ();
+	type ForceOrigin = EnsureRoot<AccountId>;
+	type Freezer = ();
+	type MetadataDepositBase = ConstU128<0>;
+	type MetadataDepositPerByte = ConstU128<0>;
+	type RemoveItemsLimit = ConstU32<1_000>;
+	type RuntimeEvent = RuntimeEvent;
+	type StringLimit = ConstU32<4>;
+	// TODO: Change before release
+	type WeightInfo = ();
```

Comment:
Can we maybe benchmark the pallet already in this PR?

Reply:
https://github.com/KILTprotocol/kilt-node/pull/692/commits/68cb4a6741c2ba2d5d4591b7f1cd4367a83ac0b1